lib/lz4.c

```diff
@@ -461,11 +461,11 @@ LZ4_FORCE_INLINE void LZ4_putPosition(const BYTE* p, void* tableBase, tableType_
     LZ4_putPositionOnHash(p, h, tableBase, tableType, srcBase);
 }
 
-static const BYTE* LZ4_getPositionOnHash(U32 h, void* tableBase, tableType_t tableType, const BYTE* srcBase)
+static const BYTE* LZ4_getPositionOnHash(U32 h, const void* tableBase, tableType_t tableType, const BYTE* srcBase)
 {
-    if (tableType == byPtr) { const BYTE** hashTable = (const BYTE**) tableBase; return hashTable[h]; }
-    if (tableType == byU32) { const U32* const hashTable = (U32*) tableBase; return hashTable[h] + srcBase; }
-    { const U16* const hashTable = (U16*) tableBase; return hashTable[h] + srcBase; }   /* default, to ensure a return */
+    if (tableType == byPtr) { const BYTE* const* hashTable = (const BYTE* const*) tableBase; return hashTable[h]; }
```

Comment:
This feels silly, but otherwise I get "`error: cast from 'const void *' to 'const unsigned char **' drops const qualifier`".

Reply:
Something a bit complex : 
in `lz4.c`, the hash table can contain either : 
- 32 bits indexes (streaming)
- 16 bits indexes (small data, 64KB, no streaming)
- 32-bits pointers (32-bits mode only)

So it's a kind of template emulation (since template are not available in C).

Because of this property, you might have to change the type of `dictHashTable` into `const void*`,
in order to avoid potential strict aliasing issues.
I believe it's a minor change with no adverse consequence in your code.

Upside is : it's allowed to cast a `void*` pointer into any other type.
Any other conversion is more dangerous, as it can be transformed into a bug when compiler's optimisation decides to use strict aliasing for some dirty performance trick.